examples/cis2-wccd-upgrade/src/lib.rs

```diff
@@ -0,0 +1,2030 @@
+
+        Some(state_address) => {
+            let unpause_time = host
+                .invoke_contract_raw(
+                    &state_address,
+                    Parameter(&[]),
+                    EntrypointName::new_unchecked("get_unpause_time"),
+                    Amount::zero(),
+                )?
+                .1;
+
+            let unpause_time = if let Some(mut unpause_time) = unpause_time {
+                unpause_time.get()?
+            } else {
+                return Err(concordium_cis2::Cis2Error::Custom(StateInvokeError));
+            };
+
+            ensure!(slot_time >= unpause_time, concordium_cis2::Cis2Error::Custom(ContractPaused));
+            // Parse the parameter.
+            let UpdateOperatorParams(params) = ctx.parameter_cursor().get()?;
+            // Get the sender who invoked this contract function.
+            let sender = ctx.sender();
+
+            let (state, state_builder) = host.state_and_builder();
+            for param in params {
+                // Update the operator in the state.
+
+                match param.update {
+                    OperatorUpdate::Add => {
+                        state.add_operator(&sender, &param.operator, state_builder)?
+                    }
+                    OperatorUpdate::Remove => state.remove_operator(&sender, &param.operator)?,
+                };
+
+                // Log the appropriate event
+                logger.log(&Cis2Event::<ContractTokenId, ContractTokenAmount>::UpdateOperator(
+                    UpdateOperatorEvent {
+                        owner:    sender,
+                        operator: param.operator,
+                        update:   param.update,
+                    },
+                ))?;
+            }
+
+            Ok(())
+        }
+    }
+}
+
+#[receive(contract = "CIS2-wCCD", name = "update_admin", parameter = "NewAdmin", mutable)]
+fn contract_update_admin<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<StateImplementation<S>, StateApiType = S>,
+) -> ContractResult<()> {
+    // Check that only the old admin is authorized to update the admin address.
+    ensure_eq!(ctx.sender(), host.state().admin, ContractError::Unauthorized);
+    // Parse the parameter.
+    let new_admin = ctx.parameter_cursor().get()?;
+    // Update admin.
+    host.state_mut().admin = new_admin;
+    Ok(())
+}
+
+#[receive(contract = "CIS2-wCCD", name = "pause", parameter = "UnpauseTime", mutable)]
+fn contract_pause<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<StateImplementation<S>, StateApiType = S>,
+) -> ContractResult<()> {
+    // Check that only the current admin can pause.
+    ensure_eq!(ctx.sender(), host.state().admin, ContractError::Unauthorized);
+    // Parse the parameter.
+    let unpause_time = ctx.parameter_cursor().get()?;
+    let slot_time = ctx.metadata().slot_time();
+    // Update unpause_time.
+    match slot_time.checked_add(unpause_time) {
```

Comment:
So the unpause_time is actually a relative time, but it is stored as an absolute time in the state.

Can you please describe the rationale for this choice? It also needs to be documented well.